packages-exp/auth-exp/src/model/id_token.d.ts

```diff
@@ -78,12 +78,14 @@ export interface APIMFAInfo {
  */
 export interface IdTokenResult {
   token: string;
-  authTime: string;
-  expirationTime: string;
-  issuedAtTime: string;
+  authTime: string | null;
+  expirationTime: string | null;
+  issuedAtTime: string | null;
   signInProvider: ProviderId | null;
   signInSecondFactor: string | null;
-  claims: {
-    [claim: string]: string;
-  };
+  claims: ParsedToken;
+}
+
+export interface ParsedToken {
+  [key: string]: string | ParsedToken;
```

Comment:
a token can include another token?

Reply:
it can include a sub object yes

Reply:
recommend naming these differently then, ParsedToken should include standardized JWT fields on it like `iat`, the nested object should be a FirebaseClaims

Reply:
Done-ish

I changed the index signature (which needs to cover all named types as well) to be string|object|undefined, the firebase specific type is then covered.